client/web/src/components/Timeline.tsx

```diff
@@ -21,15 +24,33 @@ export const Timeline: FunctionComponent<TimelineProps> = ({ stages, now, classN
         <h3>Timeline</h3>
 
         <div className={className}>
-            {stages.map((stage, stageIndex) => {
-                const previousDate = stages
-                    .map(stage => stage.date)
-                    .filter((date, index) => !!date && index < stageIndex)
-                    .pop()
-
-                return (
-                    stage.date && (
-                        <>
+            {stages
+                .map((stage, stageIndex) => ({ stage, stageIndex }))
+                .map(({ stage, stageIndex }) => {
```

Comment:
This seems like an unnecessary intermediate array?
```suggestion
                .map((stage, stageIndex) => {
```

Reply:
I need to use the index as a key but the linter doesn't let you do that, so I got around it by promoting the index to data. Thoughts?

Reply:
Why does it _need_ to be the index? Can there be a property on the stage that is unique to that stage?

It should just be a warning (not build-breaking) but generally the linter is "right" with this and shouldn't be worked around. If there _was_ a justification, an `eslint-disable` comment could be used.

Reply:
The data here is a set of steps, so the index is part of the semantics of the step. Elements are not necessarily unique here (and that causes another error at runtime, which I think is worse).

Reply:
Updated here to remove it.